geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/AttributeIdxStrategy.scala

```diff
@@ -144,7 +143,13 @@ class AttributeIdxStrategy(val filter: QueryFilter) extends Strategy with LazyLo
       singleAttrValueOnlyPlan(IndexValueEncoder.getIndexSft(sft), filter.secondary, hints.getTransform)
     } else if (IteratorTrigger.canUseAttrKeysPlusValues(descriptor.getLocalName, sft, filter.secondary, transform)) {
       // we can use the index PLUS the value
-      singleAttrPlusValuePlan(IndexValueEncoder.getIndexSft(sft), filter.secondary, hints.getTransform)
+      val indexSft = IndexValueEncoder.getIndexSft(sft)
+      val plan = singleAttrValueOnlyPlan(indexSft, filter.secondary, None)
+      val transform = hints.getTransform.map(_._2).getOrElse {
+        throw new IllegalStateException("Must have a transform for attribute key plus value scan")
+      }
+      val kvsToFeatures = attributeValueKvsToFeatures(sft, indexSft, transform, attribute)
```

Comment:
@jahhulbert-ccri I removed the AttributeKeyPlusValueIterator (to try to simplify things) and instead am adding the attribute in during kvsToFeatures on the client - it seems like that should be ok, as we weren't using the attribute for filtering in the iterator (since the scan range should handle that). Mind looking it over to verify it's ok?